mm2src/coins/coin_errors.rs

```diff
@@ -108,3 +108,10 @@ pub enum MyAddressError {
     UnexpectedDerivationMethod(String),
     InternalError(String),
 }
+
+impl std::error::Error for MyAddressError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
```

Comment:
For me implementing std::error::Error looks a bit inconsistent with our MmError error structure:
In MmError we use a different, our own approach for encapsulating embedded errors: `Vec<TraceLocation>` instead of std Error source() method.

I see that impl std::error::Error for MyAddressError is required by an external lib's ErrorReport error (which is returned by any_to_transaction_data fn).
Oh, and it was actually [my suggestion](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2330/files/209c51ec282225632d5c0e38dfe6afccfe1892e1#r1944879556) to use this ErrorReport, but it looks like it was not good: better not to use third-party errors in our functions signatures (this could be inconsistent with our error system).